kahuna/public/js/usage-rights/usage-rights-editor.js

```diff
@@ -134,70 +124,60 @@ usageRightsEditor.controller(
-            // but this isn't considered "other", it's "not set".
-            return false;
-        } else {
-            const missingVal =
-                !ctrl.getOptionsFor(property)
-                    .find(option => option === ctrl.model[property.name]);
+    ctrl.save = () => {
+        ctrl.saving = true;
+        // we save as `{}` if category isn't defined.
+        const data = ctrl.category.value ?
+            angular.extend({}, ctrl.model, { category: ctrl.category.value }) : {};
 
-            return missingVal;
-        }
+        save(data).
+        catch(uiError).
+        finally(() => ctrl.saving = false);
     };
 
-    ctrl.isRestricted = prop =>
-        ctrl.showRestrictions || ctrl.category.defaultRestrictions || prop.required;
-
-    $scope.$watch(() => ctrl.showRestrictions, onValChange(isRestricted => {
-        if (!isRestricted) {
-            delete ctrl.model.restrictions;
-        }
-    }));
-
-    function setCategory(val) {
-        ctrl.category = ctrl.categories.find(cat => cat.value === val);
-    }
-
-    function modelToData(model) {
-        const modelWithCat = angular.extend({},
-            model, { category: ctrl.category && ctrl.category.value });
-
-        return Object.keys(modelWithCat).reduce((clean, key) => {
-            // remove everything !thing including ""
-            if (modelWithCat[key]) {
-                clean[key] = modelWithCat[key];
-            }
+    ctrl.reset = () => {
+        ctrl.model = {};
+        ctrl.showRestrictions = undefined;
+    };
 
-            return clean;
-        }, {});
-    }
+    ctrl.cancel = () => ctrl.onCancel();
 
     function save(data) {
-        ctrl.error = null;
-        ctrl.saving = true;
-        $q.all(ctrl.usageRights.map((usageRights) => {
+        return $q.all(ctrl.usageRights.map(usageRights => {
             const image = usageRights.image;
             const resource = image.data.userMetadata.data.usageRights;
             return editsService.update(resource, data, image).
                 then(resource => resource.data);
-        })).catch(uiError).
-            finally(() => updateSuccess(data));
+        }));
+    }
+
+    function getUniqueCats(usageRights) {
+        return unique(usageRights.map(ur => ur.data.category));
     }
 
-    function updateSuccess(data) {
-        ctrl.model = data;
-        ctrl.onSave();
-        setCategory(data.category);
-        uiSaved();
-        ctrl.saving = false;
+    function unique(arr) {
+        return arr.reduce((prev, curr) =>
+            prev.indexOf(curr) !== -1 ? prev : prev.concat([curr]), []);
     }
 
-    function uiSaved() {
-        ctrl.saved = true;
-        $timeout(() => ctrl.saved = false, 1500);
+    // takes an array of objects and turns it into an object with an array of unique values
+    // e.g. [{ a: 1, b: 2 }, { a: 2, b: 2, c: 3 }] => { a: [1,2], b: [2], c: [3] }
+    function reduceObjectsToArrays(objects) {
```

Comment:
You may be able to use `getSetOfProperties` from `imageList` (though it uses immutable data structures rather than Object/Array)?